utils/kafkaclient/kafkaclient.go

```diff
@@ -0,0 +1,79 @@
+package kafkaclient
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/segmentio/kafka-go"
+)
+
+type KafkaClient struct {
+	connections map[string]*kafka.Writer
+	mu          sync.RWMutex
+}
+
+func NewKafkaClient() *KafkaClient {
+	return &KafkaClient{
+		connections: make(map[string]*kafka.Writer),
+	}
+}
+
+// Publish publishes a message to the specified Kafka topic.
+// It takes the Kafka host, topic, payload, and key as parameters.
+// If the connection to the Kafka host does not exist, it creates a new connection and caches it.
+// Returns an error if there was a problem publishing the message.
+func (kc *KafkaClient) Publish(kafkaHost, topic string, key []byte, value []byte) error {
+	kc.mu.Lock()
+	defer kc.mu.Unlock()
+
+	conn, err := kc.getConnection(kafkaHost)
+	if err != nil {
+		return err
+	}
+
+	msg := kafka.Message{
+		Topic: topic,
+		Key:   key,
+		Value: value,
+	}
+
+	err = conn.WriteMessages(context.Background(), msg)
+	if err != nil {
+		return fmt.Errorf("failed to publish message: %w", err)
```

Comment:
We should consider adding Error codes that we can propagate to our clients for better reasoning.